yt_dlp/extractor/espn.py

```diff
@@ -277,3 +282,121 @@ def _real_extract(self, url):
+            'https://watch-cdn.product.api.espn.com/api/product/v3/watchespn/web/playback/event?id=%s' % video_id, video_id)['playbackState']
+        title = video_data['name']
+
+        # ESPN+ subscription required, through cookies
+        if video_data.get('sourceId') == 'ESPN_DTC':
+            cookie = self._get_cookies(url).get('ESPN-ONESITE.WEB-PROD.token')
+            if not cookie:
+                raise self.raise_login_required(method='cookies')
+            id_token = cookie.value.split('|')[1]
+
+            assertion = self._call_bamgrid_api(
+                'devices', video_id,
+                headers={'Content-Type': 'application/json; charset=UTF-8'},
+                payload={
+                    'deviceFamily': 'android',
+                    'applicationRuntime': 'android',
+                    'deviceProfile': 'tv',
+                    'attributes': {},
+                })['assertion']
+
+            token = self._call_bamgrid_api(
+                'token', video_id, payload={
+                    'subject_token': assertion,
+                    'subject_token_type': 'urn:bamtech:params:oauth:token-type:device',
+                    'platform': 'android',
+                    'grant_type': 'urn:ietf:params:oauth:grant-type:token-exchange'
+                })['access_token']
+
+            assertion = self._call_bamgrid_api(
+                'accounts/grant', video_id, payload={'id_token': id_token},
+                headers={
+                    'Authorization': token,
+                    'Content-Type': 'application/json; charset=UTF-8'
+                })['assertion']
+
+            token = self._call_bamgrid_api(
+                'token', video_id, payload={
+                    'subject_token': assertion,
+                    'subject_token_type': 'urn:bamtech:params:oauth:token-type:account',
+                    'platform': 'android',
+                    'grant_type': 'urn:ietf:params:oauth:grant-type:token-exchange'
+                })['access_token']
+
+            playback = self._download_json(
+                video_data['videoHref'].format(scenario='browser~ssai'), video_id,
+                headers={
+                    'Accept': 'application/vnd.media-service+json; version=5',
+                    'Authorization': token
+                })
+
+            m3u8_url, m3u8_headers = playback['stream']['complete'][0]['url'], {'authorization': token}
+
+        # TV Provider required
+        else:
+            resource = self._get_mvpd_resource('ESPN', title, video_id, None)
+            auth = self._extract_mvpd_auth(url, video_id, 'ESPN', resource).encode('utf-8')
+
+            asset = self._download_json(
+                'https://watch.auth.api.espn.com/video/auth/media/%s/asset?apikey=uiqlbgzdwuru14v627vdusswb' % video_id,
+                video_id, data=(
+                    'adobeToken=%s&drmSupport=HLS' % compat_urllib_parse_quote_plus(base64.b64encode(auth))).encode())
```

Comment:
Do either

```suggestion
                    'adobeToken=%s&drmSupport=HLS' % urllib.parse.quote_plus(base64.b64encode(auth))).encode())
```

or 

```suggestion
                    urlencode_postdata({
                        'adobeToken': base64.b64encode(auth).decode(),
                        'drmSupport': 'HLS',
                    }))
```

And add/remove imports as needed.

Latter suggestion doesn't match parenthesis (because of L388), but better than current code